src/packages/engine-core/jest.config.js

```diff
@@ -1,14 +1,10 @@
 module.exports = {
   preset: 'ts-jest',
   testEnvironment: 'node',
-  modulePathIgnorePatterns: [
-    'build/',
-    'dist/',
-    'generator/',
-    'runtime/',
-    'scripts/',
-    'sandbox/',
-  ],
+  collectCoverage: true,
+  coverageDirectory: 'src/__tests__/coverage',
+  collectCoverageFrom: ['src/**/*.ts', '!**/__tests__/**/*', '!**/*.test.ts'],
```

Comment:
The explanation for this:

1. All source files
2. Except for anything inside tests dir (which sometimes contains non-test modules like helper modules)
3. Except all test files (some packages have `__tests__` in multiple places and it seemed this was needed to exclude them all. Thinking about it now it seems like this shouldn't be needed given rule 2 above but... meh.